common/buildcraft/lib/fluid/FluidStorage.java

```diff
@@ -0,0 +1,147 @@
+        if(fluid != null) {
+            return fluid.getFluid();
+        }
+        return null;
+    }
+
+    public String getFluidName() {
+        if(getFluidFluid() != null) {
+            return getFluidFluid().getName();
+        }
+        return "n/a";
+    }
+
+    public int getFluidColor() {
+        if(getFluidFluid() != null) {
+            if(!fluidColors.containsKey(getFluidFluid())) {
+                try {
+                    TextureMap map = Minecraft.getMinecraft().getTextureMapBlocks();
+                    String flow = getFluidFluid().getFlowing().toString();
+                    TextureAtlasSprite sprite;
+                    if(map.getTextureExtry(flow) != null) {
+                        sprite = map.getTextureExtry(flow);
+                    } else {
+                        sprite = map.registerSprite(getFluidFluid().getFlowing());
+                    }
+                    int[] pixels = sprite.getFrameTextureData(0)[0];
+                    int pixel = pixels[pixels.length / 2];
+                    // order: argb -> abgr
+                    byte[] bytes = ByteBuffer.allocate(4).putInt(pixel).array();
+                    int a = ((int) bytes[0]) & 0xFF;
+                    int r = ((int) bytes[1]) & 0xFF;
+                    int g = ((int) bytes[2]) & 0xFF;
+                    int b = ((int) bytes[3]) & 0xFF;
+                    fluidColors.put(getFluidFluid(), ((a & 0xff) << 24) + ((b & 0xff) << 16) + ((g & 0xff) << 8) + (r & 0xff));
+                } catch(Exception e) {
+                    e.printStackTrace();
+                }
+            }
+            return fluidColors.get(getFluidFluid());
+        }
+        return 0xFF_00_00_00;
+    }
+
+    @Override
+    public final NBTTagCompound writeToNBT(NBTTagCompound nbt) {
+        NBTTagCompound tankData = new NBTTagCompound();
+        super.writeToNBT(tankData);
+        nbt.setTag(name, tankData);
+        return nbt;
+    }
+
+    @Override
+    public final FluidTank readFromNBT(NBTTagCompound nbt) {
+        if (nbt.hasKey(name)) {
+            setFluid(null);
+            NBTTagCompound tankData = nbt.getCompoundTag(name);
+            super.readFromNBT(tankData);
+        }
+        return this;
+    }
+
+    @Override
+    public NBTTagCompound serializeNBT() {
+        return this.writeToNBT(new NBTTagCompound());
+    }
+
+    @Override
+    public void deserializeNBT(NBTTagCompound nbt) {
+        this.readFromNBT(nbt);
+    }
+
+    public String getDebugString() {
+        return getAmount() + " / " + capacity + " MB of " + getFluidName();
+    }
+
+    public void writeToBuffer(ByteBuf buffer) {
+        NBTTagCompound tankData = new NBTTagCompound();
+        super.writeToNBT(tankData);
+        ByteBufUtils.writeTag(buffer, tankData);
+    }
+
+    public void readFromBuffer(ByteBuf buffer) {
+        NBTTagCompound tankData = ByteBufUtils.readTag(buffer);
+        super.readFromNBT(tankData);
+    }
+}
```

Comment:
Fortunately however everyone who contributed to both "Tank" and "SingleUseTank" has agreed to re-license to MPL2 so it doesn't need to be ported :)

(Although I have only just checked that, and I hope that a lot more classes fall into this category).